### PR TITLE
Adding Docker Imaging for seamless environment setup

### PR DESCRIPTION
**Pull request by connortbot**
This branch started off as complete overhaul of the compilation by adding SDL2, a GUI viewer. I've decided not to go with this approach, as its very difficult to pull off both in code in the long run in combination with build environments. While it is possible to run GUI applications on Docker images, it is so needlessly complicated for very little payoff.

With this branch, dependencies will be managed by `vcpkg`, and compilation with `CMake` using a Docker image. 
This branch adds a `linux.sh`, which is a script to compile, and a `Dockerfile`.

It isn't fully documented yet, but I think with this branch we can move towards developing on consistent Docker images, which will get rid of our very annoying issue of cross-compilation. (Especially since we may be getting more contributors soon)

TL;DR
- No more SDL2 GUI viewer, only one mode for rendering
- Uses docker, we are not supporting Windows nor MacOS compilation unless you do it yourself (and maybe we'll support it in the future?) This follows in the footsteps of OpenMoonray.

Other changes to the wiki and documentation are needed.

I'm not sure whether we should keep the ENV scripts? Maybe since they're working for some people?

### CODE REVIEW

**Review by haenlonns — COMMENTED** [review]
Attempted to run docker command:
> docker build -t caitlyn-mcrt .

Received following error:
> ERROR: error during connect: this error may indicate that the docker daemon is not running: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/_ping": open //./pipe/docker_engine: The system cannot find the file specified.
